Assets/MixedRealityToolkit/Providers/BaseInputDeviceManager.cs

```diff
@@ -61,36 +155,97 @@ public abstract class BaseInputDeviceManager : BaseDataProvider<IMixedRealityInp
-                {
-                    var pointerProfile = InputSystemProfile.PointerProfile.PointerOptions[i];
+                var option = pointerConfigurations[i].profile;
 
-                    if (((pointerProfile.ControllerType & controllerType) != 0) &&
-                        (pointerProfile.Handedness == Handedness.Any || pointerProfile.Handedness == Handedness.Both || pointerProfile.Handedness == controllingHand))
+                if (option.ControllerType.HasFlag(controllerType) && option.Handedness.HasFlag(controllingHand))
+                {
+                    var pointerCache = pointerConfigurations[i].cache;
+                    if (pointerCache.Count > 0)
                     {
-                        var pointerObject = Object.Instantiate(pointerProfile.PointerPrefab);
-                        MixedRealityPlayspace.AddChild(pointerObject.transform);
-                        var pointer = pointerObject.GetComponent<IMixedRealityPointer>();
-
-                        if (pointer != null)
+                        var p = pointerCache.Pop();
+                        if (p != null && p is MonoBehaviour pointerGameObject)
                         {
-                            pointers.Add(pointer);
+                            Debug.Log($"{p.PointerName} retrieved from cache in request");
+
+                            pointerGameObject.gameObject.SetActive(true);
+
+                            // Track pointer for recycling
+                            activePointersToConfig.Add(p, (uint)i);
+
+                            returnPointers.Add(p);
+
+                            // We got pointer from cache, continue to next pointer option to review
+                            continue;
                         }
-                        else
+                    }
+                    
+                    // We couldn't obtain a pointer from our cache, resort to creating a new one
+                    var pointer = CreatePointer(in option);
+                    if (pointer != null)
+                    {
+                        Debug.Log($"{pointer.PointerId} created after cache miss in request");
+
+                        // Track pointer for recycling
+                        activePointersToConfig.Add(pointer, (uint)i);
+
+                        returnPointers.Add(pointer);
+                    }
+                }
+            }
+
+            return returnPointers.Count == 0 ? null : returnPointers.ToArray();
+        }
+
+        protected virtual void RecyclePointers(IMixedRealityInputSource inputSource)
+        {
+            if (inputSource != null)
+            {
+                var pointers = inputSource.Pointers;
+                for (int i = 0; i < pointers.Length; i++)
+                {
+                    var p = pointers[i];
+                    if (p != null && p is MonoBehaviour pointerComponent)
+                    {
+                        // TODO: Troy - look at reset method or other properties?
```

Comment:
>// TODO: Troy - look at reset method or other properties? [](start = 24, length = 57)

+1 It feels like there are other properties here we might want to reset (i.e. InputSourceParent, pointerName) - or maybe not? I think those will get reset to correct things once the controller is re-set the next time around, though it feels like a pointer going back into the cache should be as-close-as-possible to what happens after CreatePointer.

Reply:
Pointers may also want to reset visual state, is pressed state. Perhaps a reset() method on the IMRPointer interface which would reset to default state in case pointer gets lost, etc.